### PR TITLE
Integrate new Loss functions from Yeung et al.

### PR DESCRIPTION
**Pull request by muellerdo**
New loss functions:
- Combo loss
- Focal Tversky loss (symmetric and asymmetric)
- Focal loss (symmetric and asymmetric)
- Unified Focal loss (symmetric and asymmetric)

Reference Implementation: https://github.com/mlyg/unified-focal-loss
Manuscript: https://arxiv.org/abs/2102.04525

Includes:
- Unittesting
- Wiki Update

### CODE REVIEW

**Comment by muellerdo** [comment]
# Wiki entry:

#### Loss functions from Michael Yeung et al. Analysis

The authors Yeung et al. implemented various common and widely used loss functions as well as proposed the Unified Focal loss, a new hierarchical framework that generalises Dice and cross entropy-based losses for handling class imbalance.  

List of loss functions:
- Combo loss
- Focal Tversky loss (symmetric and asymmetric)
- Focal loss (symmetric and asymmetric)
- Unified Focal loss (symmetric and asymmetric)

Implementation: https://github.com/mlyg/unified-focal-loss  
Reference: https://arxiv.org/abs/2102.04525  

```python
from miscnn.neural_network.metrics import *

metric_list = [combo_loss,
               focal_tversky_loss,
               symmetric_focal_loss,
               symmetric_focal_tversky_loss,
               asymmetric_focal_loss,
               asymmetric_focal_tversky_loss,
               sym_unified_focal_loss,
               asym_unified_focal_loss]

model = Neural_Network(preprocessor=pp, loss=combo_loss,
                       metrics=metric_list)
```
